provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/GradleKotlinDslIntegrationTest.kt

```diff
@@ -502,6 +502,37 @@ class GradleKotlinDslIntegrationTest : AbstractIntegrationTest() {
             containsString("Settings plugin applied!"))
     }
 
+    @Test
+    fun `scripts can use the gradle script api`() {
+
+        fun usageFor(target: String) = """
+
+            logger.error("Error logging from $target")
+            require(logging is LoggingManager, { "logging" })
+            require(resources is ResourceHandler, { "resources" })
+
+            require(relativePath("settings.gradle.kts") == "settings.gradle.kts", { "relativePath(path)" })
```

Comment:
:x: I think it's important for at least one test to capture the _paths are relative to the root project dir_ semantics, did I miss one?

Reply:
I made a slight change to this assertion to capture that semantic for `settings.gradle.kts` scripts.

It doesn't really capture it for the build script case, but I don't think we need more coverage here, this is already largely covered upstream.